### PR TITLE
Add warning when running in debug mode

### LINKED ISSUE #28157: Add warning in admin page about running with debug flag
Running with "debug" flag on can lead to issues like:
- files served over remote.php/webdav are served without content-disposition attachment to make debugging easier (not sure why though, chromium can handle it fine)
- QueryLogger is going to eat up a lot of memory

@butonic @DeepDiver1975

### PR DESCRIPTION
**Pull request by PVince81**
## Description
Add warning when running in debug mode

## Related Issue
Fixes https://github.com/owncloud/core/issues/28157

## How Has This Been Tested?
Manual test

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: core/js/setupchecks.js [hunk 1]
```diff
@@ -148,6 +148,20 @@
 							type: OC.SetupChecks.MESSAGE_TYPE_ERROR
 						});
 					}
+					if (data.hasDebugMode) {
+						messages.push({
+							msg: t(
+									'core',
+									'You are running in DEBUG mode, this can cause unwanted side effects. Please remove the "debug" flag from "config.php".',
+									{
+										docLink: data.codeIntegrityCheckerDocumentation,
```
Comment (DeepDiver1975) [inline core/js/setupchecks.js]: these variables are not needed - copy and past error I assume

**Review by DeepDiver1975 — CHANGES_REQUESTED** [review]

File: core/js/setupchecks.js (hunk 1 above)
Comment (PVince81) [reply core/js/setupchecks.js]: removed, thanks for spotting

**Comment by tomneedham** [comment]
👍

**Review by tomneedham — APPROVED** [review]

**Comment by PVince81** [comment]
stable10: https://github.com/owncloud/core/pull/29936

**Pull request merged**